### ISSUE TITLE
Is there one little mistake in infogan-pytorch?

### ISSUE THREAD

**Issue by hbkunn** [wiseodd/generative-models#4]
crossent_loss = torch.mean(-torch.sum(c * torch.log(Q_c_given_x + 1e-8), dim=1))
ent_loss = torch.mean(-torch.sum(c * torch.log(c + 1e-8), dim=1))
mi_loss = crossent_loss + ent_loss

it should be "crossent_loss - ent_loss" I suppose.
the entropy should be -plog(p) (I suppose???)

if it's discontinuous, the entropy should always be zero.
but if it's continuous. it's different....

**Comment by jimzer** [comment]
I've noticed it following the the info gan code of the repo. 
I think you're right it should be "crossent_loss - ent_loss".

**Comment by leviswind** [comment]
I think you are right, but the result wouldn't be changed. Because ent_loss don't contribute to the gradient of any parmeters.
